lpcompiler.pas

```diff
@@ -871,8 +872,11 @@ function TLapeCompiler.HandleDirective(Sender: TLapeTokenizerBase; Directive, Ar
   begin
     Argument := LowerCase(Trim(Argument));
     if (Argument = 'on') or (Argument = '+') then
-      Include(FOptions, Option)
-    else if (Argument = 'off') or (Argument = '-') then
+    begin
+      Include(FOptions, Option);
+      if (Option = lcoHints) then
+        FHasHintDefine := True;
```

Comment:
Could you elaborate on why you added `FHasHintDefine`? I don't think it's really needed.

In any case, this check should not be here. If it has to be done, it has to be done where `setOption` is called. Now, every setOption call has to check whether it's for `lcoHints` while we already know the answer when we call it.